src/msw/thread.cpp

```diff
@@ -1121,6 +1122,96 @@ wxThreadError wxThread::Kill()
     return rc;
 }
 
+// At least MSVC 2017 version 15.6 is required for observing the
+// thread names set using this method.
+// Windows 10 version 1607 is required for the SetThreadDescription
+// function.
+static bool wxSetThreadNameOnWindows10(const WCHAR *threadName)
+{
+    typedef HRESULT(WINAPI* SetThreadDescription_t)(HANDLE, PCWSTR);
+    static SetThreadDescription_t s_pfnSetThreadDescription = NULL;
+
+    static bool s_initDone = false;
+    if ( !s_initDone )
+    {
+        wxLoadedDLL dllKernel32("kernel32.dll");
+        wxDL_INIT_FUNC(s_pfn, SetThreadDescription, dllKernel32);
+        s_initDone = true;
+    }
+
+    if ( s_pfnSetThreadDescription ) {
+        HRESULT r = (*s_pfnSetThreadDescription)
+            (GetCurrentThread(), threadName);
+        if ( SUCCEEDED(r) )
+        {
```

Comment:
As the statements after both `if`s are just a single line each, I would remove all 4 braces from here, the second set is surely unnecessary.